llvm/cmake/modules/AddLLVM.cmake

```diff
@@ -355,6 +349,9 @@ function(add_link_opts target_name)
     set_property(TARGET ${target_name} APPEND_STRING PROPERTY
                  LINK_FLAGS " -Wl,-brtl")
   endif()
+
+  check_linker_flag(CXX "-Wl,-Bsymbolic-functions"
```

Comment:
Add a comment what targets may need the -Wl,-Bsymbolic-functions check?

While configure is right, llvm cmake is so complex and so slow that many contributors want to optimize the number of compile/linke checks. I would still hope that we only do this on targets that actually need it.

Reply:
While I am aware of https://ewontfix.com/13/ ("Incorrect configure checks for availability of functions"), there have been many cmake changes to skip some checks that are guaranteed to succeed, to make cmake run faster (llvm's cmake is really slow). I wonder whether there is a condition that can skip the check for systems that always have -Bsymbolic-functions.

Reply:
I'd argue it's much faster than autotools for other large projects. It would be nice if the configuration phase could be parallelized, at least to some degree, but even with a single thread, it's a matter of seconds.

Reply:
@mati865 This might be true for other, smaller projects, but for LLVM we have to be careful introducing new checks. https://discourse.llvm.org/t/cmake-compiler-flag-checks-are-really-slow-ideas-to-speed-them-up/78882
It could be slower on a slow device like NFS. For this single check_linker_flag it might be fine.

Reply:
Why was this `check_linker_flag` put inside `add_linker_opts` instead of at the call site where it is actually used in the `llvm-shlib` target? This function is called *every time LLVM creates a library*. Even after the first call when the cache variable is populated, this function seems to have significant overhead.